lldb/source/Core/PluginManager.cpp

```diff
@@ -689,30 +690,38 @@ PluginManager::GetObjectFileCreateMemoryCallbackForPluginName(
 }
 
 Status PluginManager::SaveCore(const lldb::ProcessSP &process_sp,
-                               const FileSpec &outfile,
-                               lldb::SaveCoreStyle &core_style,
-                               llvm::StringRef plugin_name) {
-  if (plugin_name.empty()) {
+                               const lldb_private::CoreDumpOptions &options) {
+  Status error;
+  if (!options.GetOutputFile()) {
+    error.SetErrorString("No output file specified");
+    return error;
+  }
+
+  if (options.GetCoreDumpPluginName()->empty()) {
```

Comment:
This is a `std::optional<std::string>` and this will crash if the plug-in name is empty. There should be a test for this to make sure we handle that case correctly on mac, since that is the only platform that can save a core file using the same format as the executable (mach-o).

This line should be:

```
if (!options.GetCoreDumpPluginName().has_value()) {
```